extensions/ql-vscode/src/vscode-tests/no-workspace/eval-log-tree-builder.test.ts

```diff
@@ -0,0 +1,56 @@
+import { expect } from 'chai';
+import EvalLogTreeBuilder from '../../eval-log-tree-builder';
+import { EvalLogData } from '../../pure/log-summary-parser';
+
+describe('EvalLogTreeBuilder', () => {      
+    it('should build the log tree roots', async () => {
+        const builder = createLogTreeBuilder();
+        const roots = await builder.getRoots();
+        
+        expect(roots.map(
+          r => ({ ...r, children: undefined })
+        )).to.deep.eq(expectedRoots);
+    });
+
+    function createLogTreeBuilder() {
+        return new EvalLogTreeBuilder(evalLogDataItems); 
+    }
+
+    const evalLogDataItems: EvalLogData[] = [
+        {
+            queryCausingWork: 'test-query.ql',
+            predicateName: 'quick_eval#query#ffffffff',
+            millis: 1,
+            resultSize: 596,
+            ra: 
+                    { pipeline: [
+                        '{1} r1',
+                        '{2} r2',
+                        'return r2'
+                    ] }
+        }
```

Comment:
I'm having an issue with the formatting of the data in this integration test, again related to the ra field — the current format is what is actually passed into the tree builder and parsed (correctly); I have copied it from debugging. But this doesn't compile as `ra` expects a `Map` type rather than an object. 

When I use the following format which compiles:
```
ra: 
   new Map (Object.entries(
      { pipeline: [
           '...', '...', '...'
      ] }
    ))
```

then the code in the builder that actually parses the items is unable to parse the `Map` because it expects an `Object` that it then uses `Object.entries` on rather than a `Map`.

I'm not really sure how to resolve this issue to get the test to both run correctly and compile, but feel there should be a simple solution 🤔

Reply:
As suggested elsewhere, I think changing `Map` to `Record` in the expected interface will resolve this. `Map` is a concrete class from JS (in particular it has a hash implementation), whereas `Record` is a more permissive interface from TS that can be satisfied by any object.

Reply:
Interesting, I didn't know about `Record` — this worked! Thank you.